### ISSUE TITLE
BUG: Can't use `configurations` from command line.

### ISSUE THREAD

**Issue by ryan-roemer** [walmartlabs/eslint-config-defaults#16]
Maybe PEBCAK, but this fails for me:

```
$ eslint -c node_modules/eslint-config-defaults/configurations/walmart/es5-node.js *.js
```

with:

```
YAMLException: Cannot read config file: /Users/rye/scm/fmd/formidable-react-component-boilerplate/node_modules/eslint-config-defaults/configurations/walmart/es5-node.js
Error: JS-YAML: end of the stream or a document separator is expected at line 1, column 13:
    "use strict";
```

**Comment by pascal08** [comment]
I ran into this error too, unfortunately. What is going wrong?

**Comment by divmain** [comment]
This is because the configurations are defined as `js` files, whereas eslint expects `yaml`.  @baer, are the constituent configs composable using just the YAML syntax and `extends: ...` ?

**Comment by ryan-roemer** [comment]
@divmain @baer -- Oh yeah. So, JSON works too, and I think we have easily convertable to JSON JS at this point (now that lodash is gone). Let's convert to that!

**Comment by baer** [comment]
Yea the ESLint file reader is kinda YAML happy. If it can't figure something out about a file it just assumes it's YAML. Between that and the fact the JSON has no comments I think YAML all the way down might be a good call even though it's going to be messy. This is why I went with JS in the first place but it seems like ESLint is not really that happy with my choices :/

It's a tradeoff between
```json
  "no-unused-vars": [2, {"vars": "local", "args": "after-used"}]
```

```yaml
  # disallow declaration of variables that are not used in the code
  "no-unused-vars":
    - 2
    -
      "vars": "local"
      "args": "after-used"
```

**Comment by divmain** [comment]
Since YAML is a superset of JSON, you can also do
```yaml
  no-unused-vars: [2, {"vars": "local", "args": "after-used"}] # disallow declaration of variables that are not used in the code
```
or
```yaml
  no-unused-vars: # disallow declaration of variables that are not used in the code
    [2, {"vars": "local", "args": "after-used"}]
```
if you care about line length.

I find this to be much more readable than the YAML-arrays-on-new-lines syntax.

**Comment by pascal08** [comment]
Which files have to be converted to YAML? All .js files?

**Comment by ryan-roemer** [comment]
@pascal08 -- Anything you want to use as a CLI configuration.

@baer @divmain -- Playing around with http://nodeca.github.io/js-yaml/ we don't actually have to do that much refactoring. Here are some YAML-compliant examples:

https://github.com/walmartlabs/eslint-config-defaults/blob/master/configurations/walmart/es5-node.js

```yml
{
  "extends": [
    "./es5.js",
    "../../rules/eslint/node/walmart.js"
  ],
  "rules": {
    "strict": [2, "global"]
  }
}
```

https://github.com/walmartlabs/eslint-config-defaults/blob/master/rules/eslint/strict/walmart.js

```yml
{
  "rules": {
    # require that all functions are run in strict mode
    "strict": [2, "never"]
  }
}
```

Basically:

* Remove `module.exports` and `use strict`
* Convert `//` to `#`

... or we can go "full YAML", but even there, I'd recommend something like:

```yml
rules:
  # require that all functions are run in strict mode
  strict: [2, "never"]
```

Which is (1) keep rule arrays / scalars as straight json, (2) keep comments _above_ the rules.
